huaweicloud/services/er/resource_huaweicloud_er_route_table.go

```diff
@@ -0,0 +1,533 @@
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/er/v3/associations"
+	"github.com/chnsz/golangsdk/openstack/er/v3/propagations"
+	"github.com/chnsz/golangsdk/openstack/er/v3/routes"
+	"github.com/chnsz/golangsdk/openstack/er/v3/routetables"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceRouteTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRouteTableCreate,
+		UpdateContext: resourceRouteTableUpdate,
+		ReadContext:   resourceRouteTableRead,
+		DeleteContext: resourceRouteTableDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceRouteTableImportState,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The region where the ER instance and route table are located.`,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The ID of the ER instance to which the route table belongs.`,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The name of the route table.`,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+					validation.StringMatch(regexp.MustCompile("^[\u4e00-\u9fa5\\w.-]*$"), "The name only english and "+
+						"chinese letters, digits, underscore (_), hyphens (-) and dots (.) are allowed."),
+				),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The description of the ER route table.`,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 255),
+					validation.StringMatch(regexp.MustCompile(`^[^<>]*$`),
+						"The angle brackets (< and >) are not allowed."),
+				),
+			},
+			"tags": common.TagsForceNewSchema(),
+			// Attributes
+			"associations": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The association ID.`,
+						},
+						"attachment_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The attachment ID corresponding to the routing association.`,
+						},
+						"attachment_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The attachment type corresponding to the routing association.`,
+						},
```

Comment:
this part is same to line 109 to 123, please to make it a common schema

Reply:
done.